### PR TITLE
Add error handling for AppVeyor and globally

### LINKED ISSUE #5: AppVeyor requests are broken
The way `http` is stored means it can't be used by the client. I know this is a problem from the fact that we aren't integration testing this right now, but I've experienced it before.

The behaviour right now is that any request to AppVeyor is silently failing. It can be fixed by binding the `http` property to the `robot` before storing a reference. See [TechNottingham/hubot-hackbot/blob/master/lib/client.coffee#L9](https://github.com/TechNottingham/hubot-hackbot/blob/master/lib/client.coffee#L9)

### PR DESCRIPTION
**Pull request by Codesleuth**
Fixes #5 

Handles errors using the [hubot error handler](https://hubot.github.com/docs/scripting/#error-handling).
Handles non-200 AppVeyor responses gracefully.

### CODE REVIEW

**Comment by Blundell89** [comment]
Looks good!

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: node_js
 node_js:
   - node
-  - 4.4.5
+  - 5.11.1
```
Comment (hawx) [inline .travis.yml]: close

File: src/hubot.d.ts [hunk 2]
```diff
@@ -41,13 +41,28 @@ declare module "hubot" {
...
+  interface IHubotLogger {
+    log(levelStr: string, args: any[]);
+    error(...msg: any[]);
+    emergency(...msg: any[]);
+    alert(...msg: any[]);
+    critical(...msg: any[]);
+    warning(...msg: any[]);
+    notice(...msg: any[]);
+    info(...msg: any[]);
+    debug(...msg: any[]);
```
Comment (hawx) [inline src/hubot.d.ts]: how many levels of logging are needed?

File: src/index.ts [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
 import DeployScript from './scripts/deploy';
+import ErrorScript from './scripts/error';
 
 import { AppVeyor } from './lib/appveyor';
 import { Config } from './lib/config';
 
-module.exports = (robot: IRobot) => {
-  const appveyor = new AppVeyor(robot.http, Config.appveyor.token, Config.appveyor.account);
+module.exports = (robot: IHubot) => {
+  const appveyor = new AppVeyor(robot.http.bind(robot), Config.appveyor.token, Config.appveyor.account);
```
Comment (hawx) [inline src/index.ts]: of course :facepunch:

File: .travis.yml (hunk 1 above)
Comment (Codesleuth) [inline .travis.yml]: I noticed Heroku would not install 4.4.5 and always picked 5.11.1 - I thought v5 supported the spread operator but turns out only with `--harmony` set. NOPE.

File: src/hubot.d.ts (hunk 2 above)
Comment (Codesleuth) [inline src/hubot.d.ts]: I'm really just being concise so we can contribute these back to [Definitely Typed](https://github.com/DefinitelyTyped/DefinitelyTyped).

**Pull request merged**